### DISCUSSION TITLE
ClassVar handling on json() method in v1.8 is intented?

### DISCUSSION THREAD

**Discussion by otsuka** [pydantic/pydantic#2476] [Question] [answered]
```py
class Parent(BaseModel):
    classvar: ClassVar[str]

class Child(Parent):
    classvar = 'hello'

def test_classvar():
    c = Child()
    assert c.json() == '{"classvar": "hello"}'
```

The above test code works in pydantic v1.7.3, but does not work in v1.8.1.
In v1.8.1, `c.json() == '{}'`.

Is this behavior one of the breaking changes in v1.8 and intended one, or bug?

**Comment by PrettyWood** [comment, marked as answer]
Hi @otsuka 
Yes it's intended but you're right it doesn't have a **breaking change** mention
> allow overwriting ClassVars in sub-models without having to re-annotate them, #2061 by @layday

In v1.7.3, `Child.classvar` was considered a regular field. Now it inherits the `ClassVar` and is hence considered a class variable hence its absence in the json

**Reply by otsuka** [reply]
Thank you for your answer!

If you know, please let me know, is there a way to serialize the class variables to json?

**Reply by PrettyWood** [reply]
Hmm it's tricky.
Doesn't it work to just do
```py
class Child(Parent):
    classvar: str = 'hello'
```
to force `classvar` to be considered a regular field?

**Reply by samuelcolvin** [reply]
> Thank you for your answer!
> 
> If you know, please let me know, is there a way to serialize the class variables to json?

Or you could override the json() method.

**Reply by DrGodCarl** [reply]
Is there any consideration for adding this behavior as a parameter to `dict()` or even just supporting it in the existing `include` parameter? Overriding works okay, and maybe my usecase is niche, but it would be nice to not have to do it.
